shark-graph/src/main/java/shark/internal/HprofInMemoryIndex.kt

```diff
@@ -46,18 +41,22 @@ internal class HprofInMemoryIndex private constructor(
   private val classNames: LongLongScatterMap,
   private val objectIndex: LongObjectScatterMap<IndexedObject>,
   private val gcRoots: List<GcRoot>,
+  private val proguardMapping: ProguardMapping?,
```

Comment:
I was wondering if the mapping should happen when parsing instead? ie either passing proguardMapping to `Hprof.open` or `HprofReader.readHprofRecords`. This would provide the mapping capabilities even if you're not using the index. One issue though is that you can't tell if a string is a class name, a field name, or something else. So it probably wouldn't work?

Reply:
You're right, we need to know if it's a field or a class.
I was thinking about putting field deobfuscation in `HprofReader.readClassDumpRecord` and do class deobfuscation in `HprofReader.readHprofRecords`. But the problem is that in order to deobfuscate a class name we need to get its string id and it seems to be read independently of class data (`LOAD_CLASS` vs `STRING_IN_UTF8` in HprofReader). I guess we can't assume that that when `LOAD_CLASS` happens then its corresponding string id has already been read before, right?

Reply:
I'm not sure! Let's keep it that way for now